We get a complaint from the CI on osx (presumably with `clang`):

```
runtime/major_gc.c:1457:3: error: label followed by a declaration is a C23 extension [-Werror,-Wc23-extensions]
 1457 |   header_t marked_hd = With_status_hd(hd, marked);
```

(I suspect the easiest fix is just to inline the declaration.)

--- a/runtime/major_gc.c
+++ b/runtime/major_gc.c
@@ -1453,8 +1453,9 @@ void caml_darken_cont(value cont);
 
 Caml_inline header_t mark_header(value block, header_t hd, status marked)
 {
+  header_t marked_hd;
 again:
-  header_t marked_hd = With_status_hd(hd, marked);
+  marked_hd = With_status_hd(hd, marked);
   if (Tag_hd(hd) == Lazy_tag && Tag_hd(hd) == Forcing_tag) {
     /* To detect and mitigate a race against some other domain
      * short-circuiting alazy block, we compare-and-swap */

